azkaban-common/src/main/java/azkaban/utils/YarnUtils.java

```diff
@@ -30,6 +38,44 @@ public class YarnUtils {
   private static final String YARN_CONF_DIRECTORY_PROPERTY = "env.YARN_CONF_DIR";
   private static final String YARN_CONF_FILENAME = "yarn-site.xml";
 
+  public static final EnumSet<YarnApplicationState> YARN_APPLICATION_ALIVE_STATES = EnumSet.of(
+      YarnApplicationState.NEW,
+      YarnApplicationState.NEW_SAVING,
+      YarnApplicationState.SUBMITTED,
+      YarnApplicationState.ACCEPTED,
+      YarnApplicationState.RUNNING
+  );
+
+  public static Set<String> getAllAliveAppIDsByExecID(final YarnClient yarnClient,
```

Comment:
Clarification: any specific reason we are exposing two public methods one that returns List<AppReport> and the other that returns Set<String>? would it be any problem if just combine these methods into one?

Reply:
Good call, the one with `List<ApplicationReport>` gives full info, but the `Set<String>` is to be compatible with current kill method being used in the read-log-and-kill logic

Reply:
Refactoring is valuable, but only make sense to include after we got all the functions on the plate I think

Reply:
Thanks for the clarification.